<a href="https://colab.research.google.com/github/adrianadamata/Data_Science_Analise_Dados_Violencia_RJ/blob/master/PROJETO_2_Analisando_a_Viol%C3%AAncia_no_Rio_de_Janeirob.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<center><img src="http://sigmoidal.ai/wp-content/uploads/2019/08/logo_color.png" height="40px"></center>

# Análise de Dados da Violência no Rio de Janeiro

O histórico de violência no Estado do Rio de Janeiro destaca-se no cenário brasileiro. Sabe-se da existência de milícias e comandos paralelos. A fim de reunir informações sobre os índices de violência e tornar público, o Instituto de Segurança Pública (ISP) busca viabilizar a participação social em segurança pública no Rio de Janeiro. Ao fomentar pesquisas e análises desses dados, o ISP subsidia informações para que políticas públicas possam ser adotadas de forma mais eficiente.
Através dos dados disponilizados pelo ISP foi possível realizar um estudo sobre a violência no Estado do RJ.

## Obtenção dos Dados

Os dados foram pré-processados. Site oficial, do Governo do Estado do Rio de Janeiro: http://www.ispdados.rj.gov.br/estatistica.html

### Importando os dados


In [0]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/carlosfab/curso_data_science_na_pratica/master/modulo_02/violencia_rio.csv")
#  * Este arquivo `csv` foi pré-processado. Site oficial, do Governo do Estado do Rio de Janeiro: http://www.ispdados.rj.gov.br/estatistica.html

## Análise Inicial dos Dados

A partir do dataframe já disponível, avaliaremos as variáveis:

**Dicionário de variáveis**:
* `vano` - ano das ocorrências registradas;
* `mes ` - mês das ocorrências registradas;

**As colunas a seguir descrevem a quantidade de cada crime registrado naquele mês. Os crimes estão descritos a seguir:**
* `hom_doloso` -  homicídio doloso;
* `lesao_corp_morte` - lesão corporal seguida de morte;
* `latrocinio` 
* `hom_por_interv_policial` - homicídio por intervenção policial;
* `tentat_hom` - tentativa de homicídio;
* `lesao_corp_dolosa` - lesão corporal dolosa;
* `estupro`
* `hom_culposo` - homicídio culposo;
* `lesao_corp_culposa` - lesão corporal culposa;
* `roubo_comercio `- roubo a comércio;
* `roubo_residencia` - roubo a residência;
* `roubo_veiculo` - roubo de veículo;
* `roubo_carga` - roubo de carga;
* `roubo_transeunte` - roubo a transeunte;
* `roubo_em_coletivo` - roubo em coletivo;
* `roubo_banco` - roubo a banco;
* `roubo_cx_eletronico` - roubo a caixa eletrônico;
* `roubo_celular` - roubo de celular;
* `roubo_conducao_saque` - roubo com condução da vítima para saque;
* `roubo_apos_saque` - roubo após saque;
* `roubo_bicicleta` - roubo de bicicleta;
* `outros_roubos`
* **`total_roubos`**
* `furto_veiculos` - furto de veículos;
* `furto_transeunte` - furto a transeunte;
* `furto_coletivo` - furto coletivo;
* `furto_celular` - furto de celular;
* `furto_bicicleta` - furto de bicicleta;
* `outros_furtos`
* **`total_furtos`**
* `sequestro`    
* `extorsao` 
* `sequestro_relampago` 
* `estelionato` 
* `apreensao_drogas`
* `posse_drogas` 
* `trafico_drogas` 
* `apreensao_drogas_sem_autor`
* `recuperacao_veiculos` 
* `apf` - Auto de Prisão em Flagrante;
* `aaapai` - Auto de Apreensão de Adolescente por Prática de Ato Infracional;
* `cmp` - Cumprimento de Mandado de Prisão;
* `cmba` - Cumprimento de Mandado de Busca e Apreensão;
* `ameaca`
* `pessoas_desaparecidas` 
* `encontro_cadaver` 
* `encontro_ossada`
* `pol_militares_mortos_serv` - policiais militares mortos em serviço;
* `pol_civis_mortos_serv` - policiais civis mortos em serviço;
* **`indicador_letalidade`** 
* **`indicador_roubo_rua`** 
* **`indicador_roubo_veic`**
* **`registro_ocorrencias`** 
* **`fase`**

In [0]:
# Agora que já sabemos o que cada variável representa no dataset, vamos analisar as características do dataframe gerado no início do projeto. (df)

# A primeira característica que vamos analisar é o tamanho do dataframe (shape)
df.shape

(344, 56)

Podemos observar que são 344 entradas de períodos em que foram coletados os números dos registros de cada variável. As variáveis analisadas são as 56 colunas do dataframe.

In [0]:
# As colunas do dataframe são as variáveis analisadas acima, com seus respectivos nomes:
df.columns

Index(['vano', 'mes', 'hom_doloso', 'lesao_corp_morte', 'latrocinio',
       'hom_por_interv_policial', 'tentat_hom', 'lesao_corp_dolosa', 'estupro',
       'hom_culposo', 'lesao_corp_culposa', 'roubo_comercio',
       'roubo_residencia', 'roubo_veiculo', 'roubo_carga', 'roubo_transeunte',
       'roubo_em_coletivo', 'roubo_banco', 'roubo_cx_eletronico',
       'roubo_celular', 'roubo_conducao_saque', 'roubo_apos_saque',
       'roubo_bicicleta', 'outros_roubos', 'total_roubos', 'furto_veiculos',
       'furto_transeunte', 'furto_coletivo', 'furto_celular',
       'furto_bicicleta', 'outros_furtos', 'total_furtos', 'sequestro',
       'extorsao', 'sequestro_relampago', 'estelionato', 'apreensao_drogas',
       'posse_drogas', 'trafico_drogas', 'apreensao_drogas_sem_autor',
       'recuperacao_veiculos', 'apf', 'aaapai', 'cmp', 'cmba', 'ameaca',
       'pessoas_desaparecidas', 'encontro_cadaver', 'encontro_ossada',
       'pol_militares_mortos_serv', 'pol_civis_mortos_serv',
       'ind

Pode-se verificar o significado de cada variável no dicionário de variáveis, no início do Projeto.

In [0]:
# Os tipos das variáveis são: inteiro (int) e real (float)
df.dtypes

vano                            int64
mes                             int64
hom_doloso                      int64
lesao_corp_morte              float64
latrocinio                      int64
hom_por_interv_policial       float64
tentat_hom                      int64
lesao_corp_dolosa               int64
estupro                       float64
hom_culposo                   float64
lesao_corp_culposa            float64
roubo_comercio                  int64
roubo_residencia                int64
roubo_veiculo                   int64
roubo_carga                     int64
roubo_transeunte                int64
roubo_em_coletivo               int64
roubo_banco                     int64
roubo_cx_eletronico           float64
roubo_celular                 float64
roubo_conducao_saque          float64
roubo_apos_saque              float64
roubo_bicicleta               float64
outros_roubos                   int64
total_roubos                    int64
furto_veiculos                  int64
furto_transe

Verifica-se que as variáveis estudadas são numéricas, tipo inteiro e tipo float.

In [0]:
# visualizando as 5 primeiras linhas do dataframe:
df.head()

vano  mes  hom_doloso  ...  indicador_roubo_veic  registro_ocorrencias  fase
0  1991    1         657  ...                  1174                   NaN     3
1  1991    2         732  ...                  1097                   NaN     3
2  1991    3         713  ...                  1265                   NaN     3
3  1991    4         634  ...                  1415                   NaN     3
4  1991    5         650  ...                  1449                   NaN     3

[5 rows x 56 columns]

In [0]:
# Para identificar a porcentagem de valores ausentes de cada coluna: Na ordem decrescente
(df.isnull().sum()/df.shape[0]).sort_values(ascending = False)

roubo_bicicleta               0.802326
furto_bicicleta               0.802326
apreensao_drogas_sem_autor    0.523256
posse_drogas                  0.523256
apf                           0.523256
aaapai                        0.523256
cmp                           0.523256
trafico_drogas                0.523256
cmba                          0.523256
sequestro_relampago           0.418605
furto_coletivo                0.418605
extorsao                      0.418605
furto_celular                 0.418605
roubo_apos_saque              0.418605
roubo_conducao_saque          0.418605
roubo_cx_eletronico           0.418605
estupro                       0.418605
estelionato                   0.418605
pol_civis_mortos_serv         0.418605
pol_militares_mortos_serv     0.418605
pessoas_desaparecidas         0.383721
encontro_ossada               0.383721
roubo_celular                 0.313953
lesao_corp_culposa            0.279070
ameaca                        0.279070
sequestro                

## Análises Estatísticas da Violência no Rio de Janeiro

Através dos dados analisados de janeiro de 1991 até agosto de 2019, com a quantidade de registros de ocorrências em cada mês, podemos agora imprimir o resumo estatístico do dataframe:

In [0]:
df.describe()

vano         mes  ...  registro_ocorrencias        fase
count   344.000000  344.000000  ...            248.000000  344.000000
mean   2004.837209    6.453488  ...          53794.161290    2.994186
std       8.289036    3.447759  ...          11039.051992    0.076138
min    1991.000000    1.000000  ...          30621.000000    2.000000
25%    1998.000000    3.000000  ...          45374.500000    3.000000
50%    2005.000000    6.000000  ...          54926.500000    3.000000
75%    2012.000000    9.000000  ...          63292.500000    3.000000
max    2019.000000   12.000000  ...          75403.000000    3.000000

[8 rows x 56 columns]

Extraindo as informações das médias das variáveis a seguir:

  * `roubo_veiculo`
  * `furto_veiculos`
  * `recuperacao_veiculos`

Iremos então calcular a porcentagem de carros recuperados em relação aos carros roubados + carros furtados:
  
   $\frac{\text{recuperacao_veiculos}}{\left(\text{roubo_veiculo} + \text{furto_veiculos}\right)}$

In [0]:
rec = df.recuperacao_veiculos.mean()
roubo = df.roubo_veiculo.mean()
furto = df.furto_veiculos.mean()
rec/(roubo + furto)


0.42980346705075084

Podemos verificar que 42,98% dos veículos são recuperados.
   
Agora, os valores máximo (`max`) e mínimo (`min`) da coluna `hom_doloso`:

In [0]:
df.hom_doloso.max()


831

In [0]:
df.hom_doloso.min()

272

## Visualização de Dados

Breve contextualização...

Plote e comente os seguintes gráficos:

* Histograma de `hom_doloso`
* Gráfico de linhas para a variável `roubo_em_coletivo`


## Conclusão

Escreva suas conclusões e análises sobre os indicadores de violência do Rio de Janeiro...